src/Config.php

```diff
@@ -63,6 +66,8 @@ class Config implements ConfigInterface
 
     public function __construct(string $name = 'default')
     {
+        $this->parallelRunnerConfig = ParallelConfig::sequential();
```

Comment:
Given:

1. The audience of this tool, and the average hardware they use for local development and CI
2. As of 2024 it's basically impossible to buy a single-core single-thread CPU anymore
3. This PR has deterministic output and equal to the usual sequential run

I propose the default configuration to be set to Parallel running enabled, with 2 processes set hardcoded.

The sequential config would still be the default for single-core CPUs as for the `$this->parallelConfig->getMaxProcesses() > 1` safeguard present in the `Runner`.

Reply:
It's more about being cautious, as I don't know if I can cover negative paths correctly. Yesterday I looked at error handling and had tough time to get it right, but we'll see. We can decide when it's ready 🙂.

Also, having experience from my company, where PHPStan had to be limited in CI (parallel config) because it could not detect CPUs properly on k8s runner (was trying to utilise more than it should, which led to blocked processes that did nothing and only caused freeze of a job) I just thought it would be safer to make it as an opt-in at the beginning, and change it after some feedback from the users.

But I'm open to change it if @keradus gives green light 🙂.

Reply:
I am aware of `nproc` (and thus `fidry/cpu-core-counter`) detecting cpu numbers wrongly, indeed do not dare to enable this by default with `::detect()` :smile: 

But `2` is a safe number, and could save billions of minutes and resources all over the world when enabled by default :muscle:

Reply:
In Wirone/PHP-CS-Fixer#7 default config for parallel runner was improved, so it automatically use "detect" mode for future mode (or when `PHP_CS_FIXER_PARALLEL` is set to true-like value). As a fallback we still use sequential, but maybe we can consider running with 2 cores, @keradus?

Reply:
Let's use sequential mode as default (which is current v3 default behaviour, using existing "runner style").

If someone want's to experiment with parallel mode, we have dedicated flag for it (or setter in config), if someone want's to use full v4 experience, they will also use parallel mode.

with v4, we use ::detect() as default (and no need to hardcode 2 CPUs)